runtime-modules/project-token/src/tests/bonding_curve.rs

```diff
@@ -0,0 +1,1079 @@
+        IssueTokenFixture::default().execute_call().unwrap();
+
+        ActivateAmmFixture::default()
+            .with_linear_function_params(slope, intercept)
+            .execute_call()
+            .unwrap();
+
+        let token = Token::token_info_by_id(1);
+        assert_eq!(
+            IssuanceState::of::<Test>(&token),
+            IssuanceState::BondingCurve(BondingCurve {
+                slope,
+                intercept,
+                creator_reward
+            })
+        );
+    })
+}
+
+#[test]
+fn amm_activation_ok_with_amm_treasury_account_having_existential_deposit() {
+    let token_id = token!(1);
+    let config = GenesisConfigBuilder::new_empty().build();
+
+    build_test_externalities(config).execute_with(|| {
+        IssueTokenFixture::default().execute_call().unwrap();
+
+        ActivateAmmFixture::default().execute_call().unwrap();
+
+        let amm_treasury_account = Token::module_bonding_curve_reserve_account(token_id);
+        assert_eq!(
+            Balances::usable_balance(amm_treasury_account),
+            ExistentialDeposit::get()
+        );
+    })
+}
+
+#[test]
+fn amm_activation_ok_with_event_deposit() {
+    let token_id = token!(1);
+    let (creator_id, _) = member!(1);
+    let config = GenesisConfigBuilder::new_empty().build();
+    build_test_externalities(config).execute_with(|| {
+        IssueTokenFixture::default().execute_call().unwrap();
+        ActivateAmmFixture::default()
+            .with_linear_function_params(BONDING_CURVE_SLOPE, BONDING_CURVE_INTERCEPT)
+            .with_creator_reward(BONDING_CURVE_CREATOR_REWARD)
+            .execute_call()
+            .unwrap();
+
+        last_event_eq!(RawEvent::BondingCurveActivated(
+            token_id,
+            creator_id,
+            BondingCurve {
+                slope: BONDING_CURVE_SLOPE,
+                intercept: BONDING_CURVE_INTERCEPT,
+                creator_reward: BONDING_CURVE_CREATOR_REWARD,
+            }
+        ));
+    })
+}
+
+// --------------------- UNBONDING -------------------------------
+
+#[test]
+fn unbonding_noop_ok_with_zero_requested_amount() {
+    let (user_account_id, user_balance) = (member!(2).1, joy!(5_000_000));
+    build_default_test_externalities_with_balances(vec![(user_account_id, user_balance)])
+        .execute_with(|| {
+            IssueTokenFixture::default().execute_call().unwrap();
+            ActivateAmmFixture::default().execute_call().unwrap();
+            BondFixture::default().execute_call().unwrap();
+            let state_pre = sp_io::storage::root(sp_storage::StateVersion::V1);
+
+            let result = UnbondFixture::default()
+                .with_amount(0u32.into())
+                .execute_call();
+
+            let state_post = sp_io::storage::root(sp_storage::StateVersion::V1);
+            assert_ok!(result);
+            assert_eq!(state_pre, state_post);
+        })
+}
+
+#[test]
+fn unbond_fails_with_user_not_having_enought_crt() {
```

Comment:
After fixing the vesting leak issue, it'll be a good idea to add test for that too (ie. vesting funds cannot be used to "unbond")